### PR TITLE
Add NeoTreeWinSepartor hl

### PR DESCRIPTION
**Pull request by m-lima**
Fix https://github.com/nvim-neo-tree/neo-tree.nvim/issues/328

This PR only adds the `NeoTreeWinSeparator` so that pre 0.7 users can continue using their setup.
In the future, `NeoTreeVertSplit` could be considered for removal? And the local `winhighlight` option would override both `WinSeparator` and `VertSplit` with `NeoTreeWinSeparator`, thus allowing pre 0.7 and post 0.7 users to work

### CODE REVIEW

File: doc/neo-tree.txt [hunk 1]
```diff
@@ -863,6 +863,7 @@ NeoTreeSignColumn         |hl-SignColumn| override in Neo-tree window.
 NeoTreeStatusLine         |hl-StatusLine| override in Neo-tree window.
 NeoTreeStatusLineNC       |hl-StatusLineNC| override in Neo-tree window.
 NeoTreeVertSplit          |hl-VertSplit| override in Neo-tree window.
+NeoTreeWinSeparator       |hl-WinSeparator override in Neo-tree window.
```
Comment (cseickel) [inline doc/neo-tree.txt:866]: This is missing a closing `|` at the end of the link.

**Review by cseickel — COMMENTED** [review]
This needs to target `main` instead of `v2.x`.

File: doc/neo-tree.txt (hunk 1 above)
Comment (m-lima) [reply doc/neo-tree.txt]: 🤦

File: lua/neo-tree/setup/init.lua [hunk 2]
```diff
@@ -297,7 +297,7 @@ local function merge_global_components_config(components, config)
   return merged_components
 end
 
-local merge_renderers = function (default_config, source_default_config, user_config)
+local merge_renderers = function(default_config, source_default_config, user_config)
```
Comment (m-lima) [inline lua/neo-tree/setup/init.lua:300]: This change was just sumneko-lua auto-formatting the code. I suppose it is ok to keep it?

Comment (cseickel) [reply lua/neo-tree/setup/init.lua:300]: That's as it should be. I'm not sure how these formatting issues sneak in, but it should be without the space.

**Review by cseickel — APPROVED** [review]

**Comment by cseickel** [comment]
Thanks!

**Pull request merged**